inst/private/python_ipc_system.m

```diff
@@ -131,7 +132,13 @@
       error ('system ipc: failed to close %s (fd %d) after writing', ...
 	     tmpfilename, fd);
     end
-    [status, out] = system ([pyexec ' ' tmpfilename]);
+
+    if python_env_is_cygwin_like (pyexec)
+      converted_tmpfilename = cygpath (tmpfilename);
```

Comment:
Do you still need the original Windows path of the temporary file?
If not, you could also re-use the existing variable and simplify this portion of the code a bit.

Reply:
I mostly code in a [functional](https://en.wikipedia.org/wiki/Functional_programming) style, avoiding unnecessary mutations, maintaining [referential transparency](https://en.wikipedia.org/wiki/Referential_transparency), etc whenever possible. I would prefer using a new variable instead of overwriting old ones. (Basically, it's the same "use `const` by default" advice provided by many es6 coding styles.)

Sadly, we can't use conditional expression. Otherwise, it'd be a one-liner `converted_tmpfilename = cygpath (tmpfilename) if python_env_is_cygwin_like (pyexec) else tmpfilename`

Reply:
I guess that is a question of style and preference.
My personal preference is to avoid else branches if they "do nothing".

Reply:
No strong feelings here but in this case I'd probably do as @mmuetzel.  A few years of Python coding as also made me adverse to `else` (unless needed of course!)

Reply:
I personally find it more difficult to mentally keep track of all the places where a mutation (of binding or value) has occurred. I prefer to treat variable declarations as mathematical definitions, which cannot be changed in the middle of a proof (an example of programming in the functional style). Anyway, I've removed the else branch in this simple case as both of you find it less readable...